packages/store/src/index.ts

```diff
@@ -0,0 +1,231 @@
+import type { JSX } from "solid-js";
+import { batch, untrack } from "solid-js";
+import type { SetStoreFunction } from "solid-js/store";
+import { createStore } from "solid-js/store";
+
+type AnyFunction = (...args: any[]) => any;
+type Action<A extends AnyFunction> = (...args: Parameters<A>) => ReturnType<A>;
+type CreateAction = <A extends AnyFunction>(fn: A) => Action<A>;
+type CreateActions = <A extends object | undefined>(fns?: A) => A | undefined;
```

Comment:
This should accept only objects with functions.
If the value is `undefined` or includes something else besides functions, the user should take care of that.
The types in my playground should already be good.
```ts
type AnyFunctionsRecord = { readonly [K in string]: AnyFunction };
type Actions<T extends AnyFunctionsRecord> = { [K in keyof T]: Action<T[K]> };
function createActions<T extends AnyFunctionsRecord>(functions: T): Actions<T>
```

Reply:
I addressed your concerns but with a slightly different solution to allow store creation with getters and/or actions while also maintaining types.

```ts
type AnyFunction = (...args: any[]) => any;

type Action = <A extends AnyFunction>(fn: A) => (...args: Parameters<A>) => ReturnType<A>;
type Actions = Record<string, Action>;

export const createAction: Action =
  fn =>
  (...args) =>
    batch(() => untrack(() => fn(...args)));
export const createActions = <A extends Actions>(actions: A) => {
  let result: Actions = { ...actions };
  for (const [name, fn] of Object.entries(result)) {
    result[name] = createAction(fn);
  }

  return result as A;
};
```